eslint-plugin-aws-lsp/rules/no-workspace-folders.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * @fileoverview Rule to disallow usage of workspaceFolders property
+ */
+"use strict";
+
+module.exports = {
+  meta: {
+    type: "problem",
+    docs: {
+      description: "Disallow usage of workspaceFolders property",
+      category: "Best Practices",
+      recommended: true
+    },
+    fixable: null,
+    schema: [],
+    messages: {
+      noWorkspaceFolders: "Do not use workspaceFolders property. Use getAllWorkspaceFolders() method instead."
+    }
+  },
+
+  create: function(context) {
```

Comment:
Can we restrict the rule so that it is only valid if the `workspaceFolders` property is being accessed from an object of type `InitializeParams` or `WorkspaceFoldersInitializeParams`?